### PR TITLE
[Exchanges] init hollaex

### PR DESCRIPTION
**Pull request by GuillaumeDSM**
remaining: orders tests

requires https://github.com/Drakkar-Software/OctoBot-Trading/pull/674/files
image
image

### CODE REVIEW

File: Services/Interfaces/web_interface/models/configuration.py [hunk 1]
```diff
@@ -198,19 +202,23 @@ def _get_activation_state(name, activation_states):
...
+            parent_inspector = tentacles_management.default_parents_inspection
         klass = tentacles_management.get_class_from_string(name, abstract_class, package, parent_inspector)
         if klass:
             if with_info:
                 info = {
                     DESCRIPTION_KEY: get_tentacle_documentation(name, media_url),
                     NAME_KEY: name
                 }
-                if is_trading_mode:
+                if tentacle_type == TRADING_MODE_KEY:
```
Comment (Herklos) [inline Services/Interfaces/web_interface/models/configuration.py:217]: :+1:

File: Services/Interfaces/web_interface/models/configuration.py [hunk 2]
```diff
@@ -638,12 +646,22 @@ def get_other_exchange_list(remove_config_exchanges=False):
...
+            exchange_class = tentacles_management.get_class_from_string(
+                exchange_name, trading_exchanges.AbstractExchange,
+                exchanges,
+                tentacles_management.default_parents_inspection
+            )
+        except RuntimeError:
+            exchange_class = None
+        details[exchange_name] = {
+            "has_websockets": trading_api.supports_websockets(exchange_name, tentacles_setup_config),
+            "configurable": False if exchange_class is None else exchange_class.is_configurable()
```
Comment (Herklos) [inline Services/Interfaces/web_interface/models/configuration.py:662]: :+1:

File: Trading/Exchange/hollaex/hollaex_exchange.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+
+class hollaex(exchanges.SpotCCXTExchange):
+    DESCRIPTION = ""
+
+    DEFAULT_MAX_LIMIT = 500
+
+    def get_additional_connector_config(self):
+        urls = ccxt.hollaex().urls
+        custom_urls = tentacles_manager_api.get_tentacle_config(self.exchange_manager.tentacles_setup_config,
```
Comment (Herklos) [inline Trading/Exchange/hollaex/hollaex_exchange.py:34]: :+1:

**Review by Herklos — APPROVED** [review]
Very nice implementation :100:

**Comment by GuillaumeDSM** [comment]
everything works now with ccxt >1.69.52 ! 🎉

File: Trading/Exchange/hollaex/resources/hollaex.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+Hollaex is a basic SpotExchange adaptation for Hollaex exchange. 
+
+Change the api url to connect to a specific hollaex exchange
```
Comment (Herklos) [inline Trading/Exchange/hollaex/resources/hollaex.md:3]: 👍

**Review by Herklos — APPROVED** [review]
Great !

**Review by Herklos — APPROVED** [review]

**Comment by GuillaumeDSM** [comment]
just tested, it's working with the current ccxt version (ccxt==1.79.91) => merge and include in current release ?

**Comment by Herklos** [comment]
> just tested, it's working with the current ccxt version (ccxt==1.79.91) => merge and include in current release ?

Yes 👍

**Pull request merged**